schema/cryptography-defs.schema.json

```diff
@@ -0,0 +1,546 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://cyclonedx.org/schema/cryptography-defs.schema.json",
+  "$comment": "2025-06-30T20:56:28.764708",
+  "title": "Cryptographic Algorithm Family Definitions",
+  "description": "Enumerates cryptographic algorithm families and their specific metadata.",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
+    "$schema": {
+      "type": "string"
+    },
+    "lastUpdated": {
+      "type": "string",
+      "format": "date-time",
+      "title": "Last Updated",
+      "description": "The date and time (timestamp) when the data was last updated."
+    },
+    "algorithms": {
```

Comment:
We capture the algorithms, but do not communicate the production status.

An explicit attribute `status` with enum values such as legacy, active, evaluation, deprecated would help.

Reply:
In the same line of thought, it will be nice to communicate some known risks associated with an algorithm. Examples:


timing-attack
replay-attack
side-channel attack

It could be argued that often such risks are due to the implementations. In such cases, an example of a CBOM with components and dependencies.provides to illustrate implementation risks would be perfect.

Reply:
I think it's a good idea, but these type of attacks are very implementation-specific. Any algorithm can be implemented in a way so it is vulnerable to timing- or other side-channel attacks. So I think the crypto definitions are not the right place. Could be more applicable to vulnerability analysis.